onnxruntime/core/framework/tensor_allocator_with_mem_pattern.h

```diff
@@ -30,21 +31,30 @@ class TensorAllocatorWithMemPattern : public ITensorAllocator {
         return Status(common::ONNXRUNTIME, common::FAIL,
                       "Failed to get allocator for location: " + location.ToString());
 
-      if (mem_patterns_.patterns[i].PeakSize() > 0) {
-        void* buffer;
-        if (alloc->Info().alloc_type == OrtArenaAllocator) {
-          buffer = static_cast<IArenaAllocator*>(alloc.get())->Reserve(mem_patterns_.patterns[i].PeakSize());
-        }
-        else {
-          buffer = alloc->Alloc(mem_patterns_.patterns[i].PeakSize());
-        }
-        weights_buffers_.push_back(BufferUniquePtr(buffer, alloc));
-        auto kvp = buffers_.insert(std::make_pair(location, buffer));
-        if (!kvp.second) {
-          alloc->Free(buffer);
-          return Status(common::ONNXRUNTIME, common::FAIL, "duplicated location");
-        }
+      // Don't allocate memory when there is no memory usage..
+      if (mem_patterns_.patterns[i].PeakSize() <= 0) {
+        continue;
       }
+
+      const auto peak_size = mem_patterns_.patterns[i].PeakSize();
+      void* buffer;
+      if (alloc->Info().alloc_type == OrtArenaAllocator) {
+        // Arena has a specific way to store static memory.
+        // Arena does not reuse static memory allocated by Reserve.
+        buffer = static_cast<IArenaAllocator*>(
+          alloc.get())->Reserve(peak_size);
```

Comment:
>alloc.get())->Reserve(peak_size); [](start = 10, length = 33)

Nit: short line. no line wrap needed.

Reply:
That would exceeds 80 chars.

---
In reply to: [440499230](https://github.com/microsoft/onnxruntime/pull/4241#discussion_r440499230) [](ancestors = 440499230)

Reply:
I think the max length is 120 for ort?

(FYI: [https://github.com/microsoft/onnxruntime/blob/5e0928a777745a1536332973418dbbb7ccbf9d52/docs/Coding_Conventions_and_Standards.md](https://github.com/microsoft/onnxruntime/blob/5e0928a777745a1536332973418dbbb7ccbf9d52/docs/Coding_Conventions_and_Standards.md))

---
In reply to: [440505148](https://github.com/microsoft/onnxruntime/pull/4241#discussion_r440505148) [](ancestors = 440505148,440499230)